scripts/scripts.js

```diff
@@ -820,55 +766,88 @@ export function loadScript(url, callback, type) {
+async function loadEager() {
+  const main = document.querySelector('main');
   if (main) {
     decorateMain(main);
-    getLCPCandidate((lcpCandidateEl) => {
-      setLCPTrigger(lcpCandidateEl, async () => {
-        // post LCP actions go here
-        sampleRUM('lcp');
-
-        /* load gnav */
-        const header = document.querySelector('header');
-        const gnavPath = getMetadata('gnav') || `${getRootPath()}/gnav`;
-        header.setAttribute('data-block-name', 'gnav');
-        header.setAttribute('data-gnav-source', gnavPath);
-        loadBlock(header);
-
-        /* load footer */
-        const footer = document.querySelector('footer');
-        footer.setAttribute('data-block-name', 'footer');
-        footer.setAttribute('data-footer-source', `${getRootPath()}/footer`);
-        loadBlock(footer);
-
-        await loadBlocks(main);
-        loadCSS('/styles/lazy-styles.css');
-        addFavIcon('/styles/favicon.svg');
-
-        /* trigger delayed.js load */
-        const delayedScript = '/scripts/delayed.js';
-        const usp = new URLSearchParams(window.location.search);
-        const delayed = usp.get('delayed');
-
-        if (!(delayed === 'off' || document.querySelector(`head script[src="${delayedScript}"]`))) {
-          let ms = 3500;
-          const delay = usp.get('delay');
-          if (delay) ms = +delay;
-          setTimeout(() => {
-            loadScript(delayedScript, null, 'module');
-          }, ms);
-        }
-      });
-    });
     document.querySelector('body').classList.add('appear');
+    const lcpBlocks = ['featured-article', 'article-header'];
+    const block = document.querySelector('.block');
+    const hasLCPBlock = (block && lcpBlocks.includes(block.getAttribute('data-block-name')));
+    if (hasLCPBlock) await loadBlock(block);
+    const lcpCandidate = document.querySelector('main img');
+    const loaded = {
+      then: (resolve) => {
+        if (lcpCandidate && !lcpCandidate.complete) {
+          lcpCandidate.addEventListener('load', () => resolve());
+          lcpCandidate.addEventListener('error', () => resolve());
+        } else {
+          resolve();
+        }
+      },
+    };
+    await loaded;
+  }
+}
+
+/**
+ * loads everything that doesn't need to be delayed.
+ */
+async function loadLazy() {
+  const main = document.querySelector('main');
+
+  // post LCP actions go here
+  sampleRUM('lcp');
+
+  /* load gnav */
+  const header = document.querySelector('header');
+  const gnavPath = getMetadata('gnav') || `${getRootPath()}/gnav`;
+  header.setAttribute('data-block-name', 'gnav');
+  header.setAttribute('data-gnav-source', gnavPath);
+  loadBlock(header);
+
+  /* load footer */
+  const footer = document.querySelector('footer');
+  footer.setAttribute('data-block-name', 'footer');
+  footer.setAttribute('data-footer-source', `${getRootPath()}/footer`);
+  loadBlock(footer);
+
+  await loadBlocks(main);
+  loadCSS('/styles/lazy-styles.css');
```

Comment:
Because of the `await` on the previous line, the `lazy-styles` css is loaded after the `loadDelayed` is called. This `await` is unnecessary (the `loadBlocks` method is not async, this needs to be fixed), `loadLazy` could be turned to a sync function that only fires-and-forgets.

Reply:
well, i think the loadLazy() would need to be `async` in case you wanted to `await` for example the taxonomy or something... so i left that one in.